app/assets/javascripts/analytics/static-analytics.js

```diff
@@ -117,6 +121,19 @@
     });
   };
 
+  StaticAnalytics.prototype.setOptionsForNextPageview = function (options) {
+    if (!GOVUK.cookie || typeof options !== 'object') {
+      return;
+    }
+
+    var cookieOptions = getOptionsFromCookie();
+    $.extend(cookieOptions, options);
+
+    // Singly-stringified JSON sometimes gets escaped when put into a cookie, but inconsistently. The command-line
+    // tests will escape, but browser tests will not. Double-stringify in order to get consistently-escaped strings.
+    GOVUK.cookie('analytics_next_page_options', JSON.stringify(JSON.stringify(cookieOptions)));
```

Comment:
Yup. Done this. As discussed, this change should be robust against which version of `frontend` we use:

- the old version will attempt to use `callOnNextPage`, which will store a function in the cookie, and then delete the cookie (so the new method doesn't find it)
- the new version will store a JSON object in the cookie, which won't resolve to a function when checked by `callMethodRequestedByPreviousPage`, so it will be ignored by the old code and consumed by the new code